src/main/java/de/zalando/aruha/nakadi/exceptions/NakadiException.java

```diff
@@ -26,4 +31,11 @@ public void setProblemMessage(final String problemMessage) {
         this.problemMessage = problemMessage;
     }
 
+    public Problem asProblem() {
+        final String detail = StringUtils.isNotBlank(problemMessage) ? problemMessage : getMessage();
+        return Problem.valueOf(getStatus(), detail);
+    }
+
+    protected abstract Response.StatusType getStatus();
```

Comment:
Yes. And you are always free to catch an Exception anywhere (e.g. in the Controller) and re-throw another one or translate it differently.